File: microservices/product/service/src/main/java/fr/edjaz/microservices/core/product/ProductServiceApplication.java
package fr.edjaz.microservices.core.product;

import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan("fr.edjaz")
@Slf4j
public class ProductServiceApplication {

	public static void main(String[] args) {

		ConfigurableApplicationContext ctx = SpringApplication.run(ProductServiceApplication.class, args);

		String mongodDbHost = ctx.getEnvironment().getProperty("spring.data.mongodb.host");
		String mongodDbPort = ctx.getEnvironment().getProperty("spring.data.mongodb.port");
		LOG.info("Connected to MongoDb: " + mongodDbHost + ":" + mongodDbPort);
	}
}

File: microservices/product/service/src/main/kotlin/fr/edjaz/microservices/core/product/ProductServiceApplication.kt
package fr.edjaz.microservices.core.product

import com.fasterxml.jackson.databind.Module
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import org.slf4j.LoggerFactory
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan

@SpringBootApplication
@ComponentScan("fr.edjaz")
@EnableDiscoveryClient
class ProductServiceApplication {
    companion object {
        @Suppress("JAVA_CLASS_ON_COMPANION")
        @JvmStatic
        val logger = LoggerFactory.getLogger(javaClass.enclosingClass)
    }

    @Bean
    fun javaTimeModule(): Module {
        return JavaTimeModule()
    }
}

fun main(args: Array<String>) {
    val ctx = SpringApplication.run(ProductServiceApplication::class.java, *args)
    val mongodDbHost = ctx.environment.getProperty("spring.data.mongodb.host")
    val mongodDbPort = ctx.environment.getProperty("spring.data.mongodb.port")
    ProductServiceApplication.logger.info("Connected to MongoDb: $mongodDbHost:$mongodDbPort")
}
